classes/acrn-sblimage.bbclass

```diff
@@ -0,0 +1,88 @@
+        print("hv_cmdline: %s" % (hv_cmdline))
+        os.system("echo 'uart=mmio@0xfe042000' > %s" % (hv_cmdline))
+    else:
+        hv_cmdline = mbAcrnCmdlineDeployDir
+
+    mbAcrnBinaryDeployDirPairList = re.split(" +", d.getVar('MB_ACRN_MODULES').lstrip().rstrip())
+    print(*mbAcrnBinaryDeployDirPairList, sep = ", ")
+
+    genContainerPath = "%s/%s/slimboot/Tools/GenContainer.py" % (d.getVar('STAGING_DIR_NATIVE'), d.getVar('libexecdir'))
+
+    genContainerCmd = "%s" % d.getVar('PYTHON')
+    genContainerCmd = genContainerCmd + " %s create" % (genContainerPath)
+    genContainerCmd = genContainerCmd + " -cl"
+    genContainerCmd = genContainerCmd + " CMDL:%s" % (hv_cmdline)
+    genContainerCmd = genContainerCmd + " ACRN:%s" % (mbAcrnBinaryDeployDir)
+    counter = 0
+    for mbAcrnBinaryDeployDirPair in mbAcrnBinaryDeployDirPairList:
+        mbAcrnBinaryDeployDirPairMods = re.split(";", mbAcrnBinaryDeployDirPair.strip())
+
+        if not os.path.isfile(mbAcrnBinaryDeployDirPairMods[0]):
+            bb.error("module %s not found!" % mbAcrnBinaryDeployDirPairMods[0])
+
+        if not os.path.isfile(mbAcrnBinaryDeployDirPairMods[1]):
+            bb.error("module %s not found!" % mbAcrnBinaryDeployDirPairMods[1])
+
+        genContainerCmd = genContainerCmd + " MOD%s:%s MOD%s:%s" % (counter, mbAcrnBinaryDeployDirPairMods[0], counter+1, mbAcrnBinaryDeployDirPairMods[1])
+        counter = counter + 2
+    genContainerCmd = genContainerCmd + " -o %s/%s" % (d.getVar('WORKDIR'), d.getVar('SBLIMAGE_NAME'))
+    genContainerCmd = genContainerCmd + " -k %s" % (d.getVar('PREGENERATED_SIGNING_KEY_SLIMBOOT_KEY_SHA256'))
+    genContainerCmd = genContainerCmd + " -t MULTIBOOT"
+
+    print("genContainerCmd: %s" % (genContainerCmd))
+    os.system("%s" % (genContainerCmd))
+    os.system("install -d %s/boot; install -m 644 %s/%s %s/boot" % (d.getVar('IMAGE_ROOTFS'), d.getVar('WORKDIR'), d.getVar('SBLIMAGE_NAME'), d.getVar('IMAGE_ROOTFS')))
+}
+
+python() {
+    import re
+
+    multibootPackageDependencyList = re.split(" +", d.getVar('MB_DEPENDENCY').lstrip().rstrip())
+    multibootPackageMcDependencyList = re.split(" +", d.getVar('MB_MC_DEPENDENCY').lstrip().rstrip())
+
+    print("Multiboot variable parsing check")
+    print(*multibootPackageDependencyList, sep = ", ")
+    print(*multibootPackageMcDependencyList, sep = ", ")
+
+    for multibootPackageDependency in multibootPackageDependencyList:
+        if multibootPackageDependency:
+            dependency = "%s" % (multibootPackageDependency)
+            print("MultibootDependency: %s" % (dependency))
+            d.appendVarFlag('do_acrn_sblimage', 'depends', ' ' + dependency)
+
+    for multibootPackageMcDependency in multibootPackageMcDependencyList:
+        if multibootPackageMcDependency:
+            mcdependency = "multiconfig::%s" % (multibootPackageMcDependency)
+            print("MultibootMcDependency: %s" % (mcdependency))
```

Comment:
Perhaps some of these prints should be replaced by bb.debug?

Reply:
agree